src/hotspot/cpu/x86/c1_MacroAssembler_x86.cpp

```diff
@@ -58,11 +60,11 @@ int C1_MacroAssembler::lock_object(Register hdr, Register obj, Register disp_hdr
     movl(hdr, Address(hdr, Klass::access_flags_offset()));
     testl(hdr, JVM_ACC_IS_VALUE_BASED_CLASS);
     jcc(Assembler::notZero, slow_case);
+  } else if (LockingMode == LM_LIGHTWEIGHT) {
```

Comment:
Same question as in the aarch64 version, why is it useful to move the header-load here?

Reply:
I'll do the same unconditional preload change as in aarch64. To keep things simple.